### PR TITLE
W-16761729: add AST validation test case for propery placeholders

### PR DESCRIPTION
**Pull request by akshaysonvane**

### CODE REVIEW

File: dsl/src/test/java/org/mule/test/extension/dsl/ASTPropertyPlaceholderValidationTest.java [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+    serializer = ArtifactDeclarationXmlSerializer.getDefault(dslContext);
+  }
+
+  @Test
+  public void propertyPlaceholderValidation() {
+    String serializationResult = serializer.serialize(generateArtifactFromExtensionModels());
+    ArtifactAst ast = buildArtifactAst(serializationResult);
+    List<ValidationResultItem> errors = doValidate(ast);
+
+    assertThat(errors, iterableWithSize(0));
```
Comment (elrodro83) [inline dsl/src/test/java/org/mule/test/extension/dsl/ASTPropertyPlaceholderValidationTest.java:120]: use the emptyIterable matcher

File: dsl/src/test/java/org/mule/test/extension/dsl/ASTPropertyPlaceholderValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+    return errors;
+  }
+
+  protected ArtifactAst buildArtifactAst(final String configFile) {
+    return AstXmlParser.builder()
+        .withExtensionModels(muleContext.getExtensionManager().getExtensions())
+        .withExtensionModels(runtimeExtensionModels)
+        .withSchemaValidationsDisabled()
+        .build()
+        .parse("artifact", IOUtils.toInputStream(configFile, StandardCharsets.UTF_8));
```
Comment (elrodro83) [inline dsl/src/test/java/org/mule/test/extension/dsl/ASTPropertyPlaceholderValidationTest.java:143]: standard imports

File: dsl/src/test/java/org/mule/test/extension/dsl/ASTPropertyPlaceholderValidationTest.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+    runtimeExtensionModels = muleContext.getExtensionManager().getExtensions();
+    dslContext = DslResolvingContext.getDefault(ImmutableSet.<ExtensionModel>builder()
+        .addAll(runtimeExtensionModels)
+        .add(MuleExtensionModelProvider.getExtensionModel()).build());
+    modelResolver = DslElementModelFactory.getDefault(dslContext);
+    serializer = ArtifactDeclarationXmlSerializer.getDefault(dslContext);
+  }
+
+  @Test
+  public void propertyPlaceholderValidation() {
```
Comment (elrodro83) [inline dsl/src/test/java/org/mule/test/extension/dsl/ASTPropertyPlaceholderValidationTest.java:115]: misleading name

File: dsl/src/test/java/org/mule/test/extension/dsl/ASTPropertyPlaceholderValidationTest.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+    String serializationResult = serializer.serialize(generateArtifactFromExtensionModels());
+    ArtifactAst ast = buildArtifactAst(serializationResult);
```
Comment (elrodro83) [inline dsl/src/test/java/org/mule/test/extension/dsl/ASTPropertyPlaceholderValidationTest.java:117]: serializationResult -> appUsingAllCoreComponnets

**Review by elrodro83 — CHANGES_REQUESTED** [review]
move to EE to also test components from EE

**Comment by akshaysonvane** [comment]
Due to the move to EE, closing this PR in favor of [PR](https://github.com/mulesoft/mule-ee-integration-tests/pull/610)

**Pull request closed without merging**